File: src/types/project.ts
export type Platform = "tiktok" | "youtube_shorts" | "instagram_reels";

export interface ShortForgeProject {
  version: string;
  metadata: ProjectMetadata;
  canvas: Canvas;
  tracks: TimelineTrack[];
}

export interface ProjectMetadata {
  name: string;
  created_at: string;
  platform: Platform;
}

export interface Canvas {
  width: number;
  height: number;
  fps: number;
  duration_ms: number;
}

export interface TimelineTrack {
  id: string;
  type: "video" | "telop" | "audio";
  clips: unknown[];
}

File: src-tauri/src/models/project.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ShortForgeProject {
    pub version: String,
    pub metadata: ProjectMetadata,
    pub canvas: Canvas,
    pub tracks: Vec<super::timeline::TimelineTrack>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ProjectMetadata {
    pub name: String,
    pub created_at: String,
    pub platform: Platform,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum Platform {
    Tiktok,
    YoutubeShorts,
    InstagramReels,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Canvas {
    pub width: u32,
    pub height: u32,
    pub fps: u32,
    pub duration_ms: u64,
}
